Shared/Realm.Sync.Shared/Handles/SharedRealmHandleExtensions.cs

```diff
@@ -153,11 +171,32 @@ private static unsafe void RefreshAccessTokenCallback(IntPtr userHandlePtr, IntP
         #if __IOS__
         [ObjCRuntime.MonoPInvokeCallback(typeof(NativeMethods.SessionErrorCallback))]
         #endif
-        private static unsafe void HandleSessionError(IntPtr sessionHandlePtr, ErrorCode errorCode, sbyte* messageBuffer, IntPtr messageLength)
+        private static unsafe void HandleSessionError(IntPtr sessionHandlePtr, ErrorCode errorCode, sbyte* messageBuffer, IntPtr messageLength, IntPtr userInfoPairs, int userInfoPairsLength)
         {
             var session = Session.Create(sessionHandlePtr);
             var message = new string(messageBuffer, 0, (int)messageLength, System.Text.Encoding.UTF8);
-            Session.RaiseError(session, new SessionErrorException(message, errorCode));
+
+            SessionException exception;
+
+            switch (errorCode)
+            {
+                case ErrorCode.DivergingHistories:
```

Comment:
Should this handle the full list of four cases as per the [server docs](https://realm.io/docs/realm-object-server/#client-recovery-from-a-backup)?